wpengine/faustjs
# [MERL 350] preview support

## Description

These changes add WordPress preview support to the `faust-nx` package and it's corresponding example site.

## Testing

1. Copy `examples/next/faust-nx/.env.local.sample` to `examples/next/faust-nx/.env.local` & update to match your local WordPress site.
2. Build **faust-nx**
   ```sh
   npm run build -w faust-nx
   ```
3. Start **faust-nx-getting-started**
   ```sh
   npm run dev -w faust-nx-getting-started
   ```
4. Add some content to a page/post that has been previously published and preview
5. Notice the nextjs application resolves to /preview with preview data being displayed
6. Copy the current preview URL and paste into an incognito/private window
7. Notice the nextjs application redirect to your WordPress site to login. Once you login, you should be redirected back to the nextjs application where preview data is being shown.

## Screenshots

<img width="849" alt="Screen Shot 2022-08-15 at 2 29 11 PM" src="https://user-images.githubusercontent.com/6676674/184694067-bcd15798-4ad0-4923-a6c7-70f47fa11947.png">
